client/my-sites/comments/comment-list/index.jsx

```diff
@@ -91,13 +99,25 @@ export class CommentList extends Component {
 		}, status, [ '', '' ] );
 	}
 
+	isCommentPersisted = commentId => -1 !== this.state.persistedComments.indexOf( commentId );
+
+	// TODO: replace with array logic after persistedComments is merged
 	isCommentSelected = commentId => !! this.state.selectedComments[ commentId ];
 
-	isSelectedAll = () => this.props.comments.length === size( this.state.selectedComments );
+	isSelectedAll = () => this.getComments().length === size( this.state.selectedComments );
+
+	removeFromPersistedComments = commentId => this.setState( ( { persistedComments } ) => ( {
```

Comment:
I would suggest adding a comment or js doc here, or expanding the function a bit, since we tend not to curry too often.